ruma-events/src/reaction.rs

```diff
@@ -0,0 +1,38 @@
+//! Types for the *m.reaction* event.
+
+use crate::{
+    room::relationships::{Annotation, RelatesTo, Relation},
+    MessageEvent,
+};
+use ruma_events_macros::MessageEventContent;
+use ruma_identifiers::EventId;
+use serde::{Deserialize, Serialize};
+
+/// A reaction to another event.
+pub type ReactionEvent = MessageEvent<ReactionEventContent>;
+
+/// The payload for a `ReactionEvent`.
+#[derive(Clone, Debug, Deserialize, Serialize, MessageEventContent)]
+#[ruma_event(type = "m.reaction")]
+pub struct ReactionEventContent {
+    /// Information about related messages for
+    /// [rich replies](https://matrix.org/docs/spec/client_server/r0.6.1#rich-replies).
+    #[serde(rename = "m.relates_to")]
+    pub relates_to: RelatesTo,
```

Comment:
I don't think this should use the same data type and allow all the different forms of relations.

Reply:
Could you clarify what you mean by that a bit?

Reply:
As far as I can tell, `rel_type: "m.annotation"` is only used inside `m.reaction` events, while the other ones are only used inside `m.room.message` events. If this is correct, we should not share the `RelatesTo` type between reactions and messages¹ and instead have a dedicated type in here for the `"m.relates_to"` object as it appears in reactions.

¹ thus also moving the existing one back into `room.rs`

Reply:
Ah, that's what you meant, the problem with that is that there may be future uses for `m.annotation` that will be inside `m.room.message` events also `m.refrerence` is already used for in-room verifications as can be seen over [here](https://github.com/uhoreg/matrix-doc/blob/e2e_verification_in_dms/proposals/2241-e2e-verification-in-dms.md#accepting-a-key-verification).

We might paint ourselves into a corner if we use different types for reactions vs the rest here, but I'm not that against it, though moving the existing ones back into `room.rs` doesn't make sense considering that the verification events use them.

Reply:
E2E verification in DMs is a whole different can of worms. My gut feeling is that that too should have its own type for `m.relates_to` since it seems extremely unlikely a non `rel_type` different from `m.reference` would ever be allowed there.

Reply:
Fine by me then, no the verification events probably won't change for quite some time.